### DISCUSSION TITLE
A problem when simulating with the option -F or --fast-forward

### DISCUSSION THREAD

**Discussion by LiyanChen-SJTU** [gem5/gem5#1335] [Q&A] [answered]
Hi, all! I'm trying to using gem5 (version 23.1) to simulate my workload and encounter with the problem while using option -F or --fast-forward.  Then I leverage the se.py to test:

command line: build/X86/gem5.opt configs/deprecated/example/se.py --cmd=tests/test-progs/hello/bin/x86/linux/hello -F 10000

warn: The se.py script is deprecated. It will be removed in future releases of  gem5.
AttributeError: 'tuple' object has no attribute 'numThreads'

At:
  configs/deprecated/example/se.py(180): <module>
  src/python/m5/main.py(669): main

I don't know how to use the option of fast-forward. 
Any help on this would be very helpful! Thank you in advance!

**Comment by zybzzz** [comment]
Hello! I tried to reproduce your problem, but I can't find the -F or --fast-forward option either in main.py or se.py, could you please provide details?

**Reply by LiyanChen-SJTU** [reply]
Hello, thanks for your help! The -F or --fast-forward option can be shown by using the following command line.

command line:  ./build/X86/gem5.opt configs/deprecated/example/se.py -h

And the option is between the warmup-insts and --simpoint.

  -W WARMUP_INSTS, --warmup-insts WARMUP_INSTS
                        Warmup period in total instructions (requires --standard-switch)
  --bench BENCH         base names for --take-checkpoint and --checkpoint-restore
  **_### -F FAST_FORWARD, --fast-forward FAST_FORWARD
                        Number of instructions to fast forward before switching_**
  -S, --simpoint        Use workload simpoints as an instruction offset for --checkpoint-restore or --take-checkpoint.
  --at-instruction      Treat value of --checkpoint-restore or --take-checkpoint as a number of instructions.

Thank you in advance!

**Reply by zybzzz** [reply, marked as answer]
I looked at your code and the problem is that the `CPUClass` returned is a tuple and not a class instance, which causes the error to be reported. Please add `CPUClass = CPUClass[0]` before `CPUClass.numThreads = numThreads` and it will solve the problem, at least for me.

Also se.py is deprecated, try not to use it, the file looks like it hasn't been maintained for a long time.

Hope this helps.

**Reply by LiyanChen-SJTU** [reply]
Thank you for your answer and help!
